src/FuseGPUThreadLoops.cpp

```diff
@@ -203,7 +204,7 @@ class ReplaceForWithIf : public IRMutator {
 
             Stmt body = mutate(op->body);
 
-            Expr var = Variable::make(Int(32), "." + thread_names[dim]);
+            Expr var = Variable::make(Int(32), gpu_thread_name(dim));
```

Comment:
Should this be a UInt(32) ?

Reply:
I don't think so. Our loop variables are currently always Int(32).